calico/network-policy/policy-rules/log-rules.mdx

```diff
@@ -0,0 +1,191 @@
+
+![curl](/img/calico/curl-bpf-log.svg)
+
+1. Process name and ID.
+   Be aware that the process name might not directly correspond to the responsible process for the traffic.
+2. CPU number, Kernel flags, timestamp, BPF print function used to log this message.
+3. Interface name.
+4. Traffic type.
+   In this example, egress (`E`= egress, `X`= XDP, `I`= ingress).
+5. Policy verdict (`ALLOWED`, `DENIED`).
+6. IP protocol number.
+   In this example, 6 is for TCP (1 ICMP, 17 UDP).
+7. Source IP and source port.
+8. Destination IP and destination port.
+
+## Prerequisites
+
+* Your cluster is not a containerized Kubernetes environment such as Kind.
+* If you're running $[prodname] with the eBPF dataplane, your cluster node or VM host is running on a Linux distribution based on the Linux kernel 5.16 or later.
+
+
+
+
+## Log all traffic for an unprotected cluster
+
+Log rules can help you write network policy by showing you your cluster's current traffic.
+When you know what traffic you want to allow, you can write policies to allow that traffic and restrict all other traffic.
+
+* To log all traffic across the entire cluster, create a `GlobalNetworkPolicy` resource with log actions for ingress and egress:
+
+  ```yaml
+  apiVersion: projectcalico.org/v3
+  kind: GlobalNetworkPolicy
+  metadata:
+      name: <log-everything-in-cluster>
+  spec:
+      order: 100000
+      tier: default
+      types:
+      - Ingress
+      - Egress
+      egress:
+      - action: Log
+      ingress:
+      - action: Log
+  ```
+
+## Log traffic for a namespace
+
+To monitor or troubleshoot traffic within a namespace, use a `NetworkPolicy` resource.
+Apply the policy with the highest priority to capture traffic not handled by existing policies.
+This provides visibility into unhandled traffic patterns specific to that namespace.
+
+* To log all traffic within a specific namespace, create a NetworkPolicy like this:
+
+  ```yaml
+  apiVersion: projectcalico.org/v3
+  kind: NetworkPolicy
+  metadata:
+    name: log-everything-in-ns
+    namespace: <namespace>
+  spec:
+    order: 100001
+    tier: default
+    types:
+    - Ingress
+    - Egress
+    egress:
+    - action: Log
+    ingress:
+    - action: Log
+  ```
+
+## Log traffic for non-namespaced resources, hosts, and VMs
+
+To use log rules for non-namespaced resources, hosts, and VMs, you can create host endpoints for them.
+By default, $[prodname] policies apply only to namespaced resources (for example, Pods, ServiceAccounts).
+
+1. Configure $[prodname] to automatically create host endpoints by running the following command:
+
+   ```bash
+   kubectl patch kubecontrollersconfiguration default --type=merge --patch='{"spec": {"controllers": {"node": {"hostEndpoint": {"autoCreate": "Enabled"}}}}}'
+   ```
+
+1. Create a log rule for testing that includes host endpoints.
+
+1. When you have completed testing, remove the log policies.
+   Leaving them in place can significantly affect cluster performance.
```

Comment:
This proviso was attached to this section on host endpoints. But is this general workflow something that should apply to all log rules? 

Are there cases where we want to have log rules set continuously? Or are they only to be used as part of a workflow that goes: 1) create log rule 2) see what happens, make fixes 3) remove log rule when you're done.